common/OpTestSystem.py

```diff
@@ -362,11 +362,11 @@ def goto_state(self, state):
         # if user overrides from command line and machine not at desired state can lead to exceptions
         self.block_setup_term = 1  # block in case the system is not on/up
         self.target_state = state  # used in WaitForIt
-        if (isinstance(self.console, OpTestQemu.QemuConsole)
-                or isinstance(self.console, OpTestMambo.MamboConsole)) \
-                and (state == OpSystemState.OS):
-            raise unittest.SkipTest(
-                "OpTestSystem running QEMU/Mambo so skipping OpSystemState.OS test")
+        if state == OpSystemState.OS:
+            if (isinstance(self.console, OpTestQemu.QemuConsole)):
+                    raise unittest.SkipTest("OpTestSystem running QEMU/Mambo so skipping OpSystemState.OS test")
+            elif (isinstance(self.console, OpTestMambo.MamboConsole)):
+                self.state = OpSystemState.OS
```

Comment:
Isn't this going to break if we're not already in the petitboot shell state?

Reply:
Oh and can you fix up the `QEMU/Mambo` comment to just say QEMU? We should probably look at enabling better testing in QEMU since it has pnor support and you can boot to a real OS if you want, but that's a problem for another PR.